src/main/scala/uk/gov/nationalarchives/consignmentexport/BagMetadata.scala

```diff
@@ -47,7 +47,7 @@ class BagMetadata(keycloakClient: KeycloakClient)(implicit val logger: SelfAware
     val contactName = getContactName(consignment.userid)
 
     Map(
-      InternalSenderIdentifierKey -> consignment.consignmentReference,
+      InternalSenderIdentifierKey -> Some(consignment.consignmentReference),
```

Comment:
Under what circumstances is this optional? I think it's a non-optional field in the API response.

Reply:
Never. This is an interim map with signature `Map[String, Option[String]]` to retrieve consignment information via the API. Some of which is an option, other, is not, for example the consignment reference. This map is then used to generate the consignment information for the bag-info.txt, or throw an exception if any of the information is missing.

Reply:
How did it compile before this change? Ah, was it optional in the generated GraphQL until now?

Reply:
Yes. The GraphQl query was updated when Loki made her changes. Those changes just weren't used yet in the export repo

Reply:
👍